### PR TITLE
Add ruby client doc

### LINKED ISSUE #1834: Document: ruby client
https://github.com/risingwavelabs/risingwave/pull/14859

cc: @xuefengze 

could you mimic the documentation for other clients, e.g. https://docs.risingwave.com/docs/current/python-client-libraries/, to give a few examples for the ruby client?

If there is any limitation, please notify users, thanks

### PR DESCRIPTION
**Pull request by xuefengze**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1834

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by hengm3467** [comment]
@ShanlanLi @WanYixian Please take a look at it. Thanks.

File: docs/dev/ruby-client-libraries.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+---
+id: ruby-client-libraries
+title: Use RisingWave in your Ruby application
+description: Use RisingWave in your Ruby application
```
Comment (ShanlanLi) [inline docs/dev/ruby-client-libraries.md]:
```suggestion
description: Use RisingWave in your Ruby application.
```

File: docs/dev/ruby-client-libraries.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+```ruby
+require 'pg'
+
+conn = PG.connect(host: '127.0.0.1', port: 4566, dbname: 'dev', user: 'root')
+```
+
+:::note
+
+`BasicTypeMapForResults` class doesn't support currently, you need cast RisingWave types into Ruby types manually.
```
Comment (ShanlanLi) [inline docs/dev/ruby-client-libraries.md]:
```suggestion
The `BasicTypeMapForResults` class isn't supported currently, you need to cast RisingWave types into Ruby types manually.
```

**Review by ShanlanLi — APPROVED** [review]
LGTM, just did minor language modification.

**Comment by WanYixian** [comment]
Currently, the table of contents looks like this:
image
Since this guide only mentions one driver, shall we change all the title levels into ##, just like the "Go" application below? Thanks!
image

**Review by WanYixian — APPROVED** [review]
LGTM!

**Pull request merged**